components/compliance-service/ingest/ingestic/ingestic.go

```diff
@@ -192,9 +324,67 @@ func (backend *ESClient) setDailyLatestToFalse(ctx context.Context, nodeId strin
 }
 
 func (backend *ESClient) DeleteProjectTag(ctx context.Context,
-	projectTagToBeDelete string) ([]string, error) {
-	// TODO implement
-	return []string{}, nil
+	projectTagToDelete string) ([]string, error) {
+	logrus.Debugf("starting project delete project %q", projectTagToDelete)
+
+	esReportJobID, err := backend.DeleteReportProjectTag(ctx, projectTagToDelete)
+	if err != nil {
+		return []string{}, errors.Wrap(err, "Failed to start Elasticsearch Report project tags delete")
+	}
+
+	esSummaryJobID, err := backend.DeleteSummaryProjectTag(ctx, projectTagToDelete)
+	if err != nil {
+		return []string{}, errors.Wrap(err, "Failed to start Elasticsearch Summary project tags delete")
+	}
+
+	logrus.Debugf("Started Project delete with report job ID: %q, summary job ID %q", esReportJobID, esSummaryJobID)
+
+	return []string{esReportJobID, esSummaryJobID}, nil
```

Comment:
As part of the signature change, instead of returning the two IDs here, then need to add a monitoring loop like that shown here: https://github.com/chef/automate/pull/2428/files#r354108737

Given it would be the same monitoring loop, sounds like a common helper is called for.

(I am happy to handle those changes in #2389 if this is merged as is, first.)

Reply:
I'm into the rule of 3s when DRYing things up . . . will there be 3? I've lost track